Add unit tests for DateFormater, DecimalFormater and MemRateFormater

petersear opened the pull request:
Hi, I've analysed your codebase and seen some gaps in the coverage of:

org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.DateFormater
org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.DecimalFormater
org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.MemRateFormater

I've written tests with the help of [Diffblue Cover](https://www.diffblue.com/products) and increased line coverage for these subclasses from 0% to 100% (using JaCoCo). Hopefully, they will help you detect regressions caused by future code changes.

freddytuxworth commented on sjk-core/src/test/java/org/gridkit/jvmtool/cmd/StackSampleAnalyzerCmdTest.java:
```diff
@@ -0,0 +1,49 @@
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testToStringDateFormater() {
```
To my mind, `testDateFormaterToString` would make more sense as the method name format (like `DateFormater.toString` but without the dots).

petersear replied:
Agreed, thanks. Resolved this.

freddytuxworth commented on sjk-core/src/test/java/org/gridkit/jvmtool/cmd/StackSampleAnalyzerCmdTest.java:
```diff
@@ -0,0 +1,49 @@
+
+  @Test
+  public void testToStringDateFormaterException() {
+    thrown.expect(NullPointerException.class);
```
If you have just one test that uses `thrown.expect`, you can have it as the last test in a method with other tests (i.e. these two lines could be appended to `testToStringDateFormater`). Again, maybe it's clearer as it is, I'm not sure.

petersear replied:
I've combined them and it makes the file neater.

sudo-jaa commented on sjk-core/src/test/java/org/gridkit/jvmtool/cmd/StackSampleAnalyzerCmdTest.java:
```diff
@@ -0,0 +1,49 @@
+import org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.DecimalFormater;
+import org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.MemRateFormater;
+import org.junit.Assert;
+import org.junit.Rule;
```
Do we need this Rule import if we already import org.junit.rules.ExpectedException?

petersear replied:
It is needed for `thrown`.

sudo-jaa commented on sjk-core/src/test/java/org/gridkit/jvmtool/cmd/StackSampleAnalyzerCmdTest.java:
```diff
@@ -0,0 +1,49 @@
+    Assert.assertEquals("70m/s", new MemRateFormater().toString(74160839));
+    Assert.assertEquals("10g/s", new MemRateFormater().toString(10737418240l));
+  }
+
```
Kill line 48 blank line?

petersear replied:
I left it in for symmetry :-)
The class opens and ends with a blank line.
There is a blank line before and after each test block.

sudo-jaa reviewed (approved):
Only minor pedantic comments, otherwise I can't see anything I would change!

Commit:
Add unit tests for DateFormater, DecimalFormater and MemRateFormater

Unit tests added for:
org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.DateFormater
org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.DecimalFormater
org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.MemRateFormater

These tests were written using Diffblue Cover.

petersear commented:
I've pushed a new commit with test names changed as suggested and adding the thrown exception to the asserts for that method, testDateFormaterToString()

peterschrammel commented on sjk-core/src/test/java/org/gridkit/jvmtool/cmd/StackSampleAnalyzerCmdTest.java:
```diff
@@ -0,0 +1,45 @@
+package org.gridkit.jvmtool.cmd;
+
+import org.gridkit.jvmtool.cmd.StackSampleAnalyzerCmd.DateFormater;
```
Oups, are there really typos in their class names?

petersear replied:
Yes, that's right.
